mysql-test/suite/rpl/t/rpl_slave_max_statement_time.test

```diff
@@ -0,0 +1,166 @@
+#
+# Purpose:
+#   This test ensures that the slave can limit the execution time of its
+# events via the variable slave_max_statement_time. Additionally, it ensures
+# that the status variable Slave_statement_timeouts shown by
+# SHOW ALL SLAVES STATUS reflects the number of events that timed out.
+#
+# Methodology:
+#   This test uses the following test cases to ensure that a slave will
+# correctly limit the execution time of its events:
+#   1) Using a serial slave, the SQL thread should time out when its underlying
+#      event executes for longer than slave_max_statement_time and increment
+#      Slave_statement_timeouts by one.
+#   2) The slave status variable Slave_statement_timeouts should continually
+#      increment after each timed out event.
+#   3) Using a parallel slave, a worker thread should time out when its
+#      underlying event executes for longer than slave_max_statement_time.
+#   4) Events should time out when waiting for a lock.
+#   5) Locally executed long running statements should not time out due to
+#      slave_max_statement_time.
+#   6) Reset slave should preserve the value of Slave_statement_timeouts.
+#   7) slave_max_statement_time should not be allowed to change while the
+#      slave is running.
+#
+# References:
+#   MDEV-27161: Add option for SQL thread to limit maximum execution time per
+#               query replicated
+#
+
+--source include/master-slave.inc
+
+--echo # Use statement format so the slave executes SLEEP functions originally
+--echo # executed on the master
+--source include/have_binlog_format_statement.inc
+
+# Valgrind interferes with thread scheduling
+--source include/not_valgrind.inc
+
+--echo #
+--echo # Set up
+--echo #
+--connection master
+SET sql_log_bin=0;
+CALL mtr.add_suppression("Unsafe statement written to the binary log using statement format");
```

Comment:
I'd recommend `SET STATEMENT sql_log_bin=0 FOR  call ...` (that way you'd never be waken up by a programmer's classic nightmare of recalling or "recalling" unmatched the opened "parenthesis" :scream: )